loadgenerator/loadgen

```diff
@@ -400,6 +399,45 @@ def _check_prereqs():
     except subprocess.CalledProcessError:
         return False
 
+#####################################
+# CLOUD RUN LOAD GENERATOR COMMANDS #
+#####################################
+
+def run_auto(args):
+    """Create automatic load generator and Web UI for normal traffic pattern.
+    Use Cloud Run to host the container and serve the Locust Web UI.
+    args.target_address -- the IP address to send load (frontend)
+    """
+    # _setup_cloud_run()
+    name = _generate_job_name()
+    logger.info('Creating load generation job {0}...'.format(name))
+
+    if not _is_address_accessible(args.target_address):
+        logger.error('Could not access target address {0}'.format(args.target_address))
+        print('Please double check the target address and try again.')
+        return
+
+    command = (
+        'gcloud run deploy {0}'
+        ' --image gcr.io/stackdriver-sandbox-230822/sandbox/loadgenerator:gz'
+        ' --update-env-vars="FRONTEND_ADDR={1}"'
+        ' --allow-unauthenticated'
+    ).format(name, args.target_address)
+
+    subprocess.call(['echo'] + split(command))
+    try:
+        output = subprocess.check_output(split(command)).decode('utf-8')
+        print(output)
+    except subprocess.CalledProcessError:
+        return
+
+def _setup_cloud_run():
+    """Configure Cloud Run with default settings so that the user isn't
+    prompted every time a command runs.
+    """
+    logger.info('Setting default platform and region')
+    subprocess.run(split('gcloud config set run/platform managed'))
+    subprocess.run(split('gcloud config set run/region us-central1'))
```

Comment:
Not sure what is the advantage of having this wrapped in python script as opposed to being a bash script. Could you elaborate?

Reply:
I guess I'm just more familiar with python. Don't think there's any advantage/disadvantage